kratos/utilities/normal_calculation_utils.cpp

```diff
@@ -318,6 +388,61 @@ void NormalCalculationUtils::CalculateNormal3D(
     rCondition.SetValue(NORMAL, rAn);
 }
 
+/***********************************************************************************/
+/***********************************************************************************/
+
+void NormalCalculationUtils::CalculateNormalShapeDerivative3D(
+    ConditionType& rCondition
+    )
+{
+    KRATOS_TRY
+
+    Geometry<Node<3>>& rGeometry = rCondition.GetGeometry();
```

Comment:
```suggestion
    const auto& rGeometry = rCondition.GetGeometry();
```

looks to me like it can be `const`